diff --git a/arch/arm64/boot/dts/arm/qemu_armv8a.dts b/arch/arm64/boot/dts/arm/qemu_armv8a.dts
index 783fa0ce..95c7095f 100644
--- a/arch/arm64/boot/dts/arm/qemu_armv8a.dts
+++ b/arch/arm64/boot/dts/arm/qemu_armv8a.dts
@@ -314,38 +314,39 @@
 	};
 
 	flash@0 {
-		bank-width = <0x4>;
-        #address-cells = <0x2>;
+		bank-width = <0x2>;
+        #address-cells = <0x1>;
         #size-cells = <0x1>;
-		reg = <0x0 0x0 0x0 0x4000000 0x0 0x4000000 0x0 0x4000000>;
+		reg = <0x0 0x0 0x0 0x4000000>;
 		compatible = "cfi-flash";
-        partition@0 {
+
+        partition@uboot {
             /* 1MB u-boot, 4 sectors */
             reg = <0x00000000 0x00100000>;
             label = "uboot";
         };
-        partition@1 {
+        partition@uboot-env {
             /* 1MB, 4 sector */
-            reg = <0x00100000 0x00040000>;
+            reg = <0x00100000 0x00100000>;
             label = "uboot-env";
-            env_size = <0x2000>;
+            env_size = <0x100000>;
         };
-        partition@2 {
+        partition@board_eeprom {
             /* 256KB, 1 sector */
-            reg = <0x00200000 0x00040000>;
+            reg = <0x00200000 0x00100000>;
             label = "board_eeprom";
         };
-        partition@3 {
+        partition@blank {
             /* 256KB, 1 sector */
-            reg = <0x00240000 0x00dc0000>;
+            reg = <0x00300000 0x00d00000>;
             label = "blank";
         };
-        partition@4 {
+        partition@onie {
             /* 16MB onie */
             reg = <0x01000000 0x01000000>;
             label = "onie";
         };
-	};
+    };
 
 	cpus {
 		#size-cells = <0x0>;
